contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,77 @@
+# Priority in ResourceQuota
+
+@resouer @bsalamat @davidopp
+
+Aug 2017
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Expected behavior of ResourceQuota apply to pod](#resourcequota-apply-to-pod)
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make ResourceQuota become priority aware, three sub-tasks are included.
+
+* Add a field to ResourceQuota that specifies a priority class name.
+* Incorporate priority in quota checks.
+* ResourceQuota admission controller to check ResourceQuota priority class and resolve the priority class to its integer value.
+
+### Non-Goals 
+
+* Add priority in Pod spec (this is implemented in: #45610)
+* Incorporate priority in pod scheduling logic.
+
+## Background 
+
+Since we already have [priority filed in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610), 
+Pods can now be classified into different priority classes. So it is nature to make ResourceQuota apply to Pods based on its priority class. In order to implement this, we need to add a priority class name filed to ResourceQuota definition.
```

Comment:
Please replace "So it is nature to make ResourceQuota apply to Pods based on its priority class." by "We would like to be able to create quota for various priority classes in order to manage cluster resources better and limit abuse scenarios."